drools-ansible-rulebook-integration-main/src/test/java/org/drools/ansible/rulebook/integration/main/PerfTest.java

```diff
@@ -12,6 +15,19 @@ public void testManyEvents() {
         checkDuration("100k_event_rules_ast.json", 10_000);
     }
 
+    @Ignore("Disabled by default, because it takes around 40 seconds")
+    @Test
+    public void testManyLargeEvents() {
+        // match_multiple_rules: false means events are removed after match. So this test will pass without throwing MemoryThresholdReachedException
+        checkDuration("1m_event_with_20kb_payload_rules_ast.json", 120_000);
+    }
+
+    @Test
+    public void testManyLargeEventsMatchMultipleRules() {
+        // match_multiple_rules: true means events are retained until TTL expires
+        assertThrows(MemoryThresholdReachedException.class, () -> checkDuration("1m_event_with_20kb_payload_match_multiple_rules_ast.json", 120_000));
+    }
```

Comment:
With this test, we can confirm MemoryThresholdReachedException is thrown before hitting OutOfMemoryError.